.github/workflows/ci-core.yml

```diff
@@ -22,21 +22,17 @@ jobs:
     steps:
     - uses: actions/checkout@v4
 
-    - name: Set up Python
-      uses: actions/setup-python@v5
+    - name: Install uv
+      uses: astral-sh/setup-uv@v3
       with:
-        python-version: ${{ matrix.python-version }}
-        cache: 'pip' # caching pip dependencies
-        cache-dependency-path: |
-          setup.py
-          requirements.txt
+        enable-cache: true
+        cache-dependency-glob: "uv.lock"
 
-    - name: Upgrade pip
-      run: |
-        pip install --upgrade pip
+    - name: Set up Python ${{ matrix.python-version }}
+      run: uv python install ${{ matrix.python-version }}
 
-    - name: Install the package and dev dependencies locally
-      run: pip install -e ".[dev]"
+    - name: Install the project
+      run: uv sync --all-extras
```

Comment:
this one should just install the dev requirements

Reply:
Fixed by https://github.com/xdslproject/xdsl/pull/3294/commits/cc1743df8a6d9c538f557fc66fcd281edaf9e2f2 switching to the `--only-dev` flag